include/map.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef __MAP_HPP__
+#define __MAP_HPP__
+
+#include "bst.hpp"
+
+template <class K, class V>
+class Map {
+public:
+    typedef BinarySearchTree<std::pair<K,V>>::iterator iterator;
+    
+    Map();
+    Map(const std::initializer_list<std::pair<K, V>> &pairs);
+    ~Map();
+
+    Map(const map<K, V> &other);
+    operator=(const map<K, V> &other);
+
+    const V& operator[](const K &key) const;
+    V& operator[](const K &key);
+
+    const V& at(const K &key) const;
+    V& at(const K &key);
+
+    std::pair<iterator, bool> insert(const std::pair<K, V> &value); 
+    size_t erase(const K &key);
```

Comment:
`erase` method return type should be `bool` instead of `size_t` so it is not clear what should return it, e.g.: size? which is element removed? status?